platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -126,6 +130,313 @@ uint32_t FreeDesktopPortalDesktop::get_appearance_color_scheme() {
+	uint8_t uuid[64];
+	Error rng_err = rng.get_random_bytes(uuid, 64);
+	ERR_FAIL_COND_V_MSG(rng_err, rng_err, "Failed to generate unique token.");
+
+	fd.connection = dbus_bus_get(DBUS_BUS_SESSION, &err);
+	if (dbus_error_is_set(&err)) {
+		dbus_error_free(&err);
+		unsupported = true;
+		ERR_FAIL_V_MSG(FAILED, vformat("Failed to open DBus connection: %s", err.message));
+	}
+
+	String dbus_unique_name = String::utf8(dbus_bus_get_unique_name(fd.connection));
+	String token = String::hex_encode_buffer(uuid, 64);
+	String path = vformat("/org/freedesktop/portal/desktop/request/%s/%s", dbus_unique_name.replace(".", "_").replace(":", ""), token);
+
+	fd.path = vformat("type='signal',sender='org.freedesktop.portal.Desktop',path='%s',interface='org.freedesktop.portal.Request',member='Response',destination='%s'", path, dbus_unique_name);
+	dbus_bus_add_match(fd.connection, fd.path.utf8().get_data(), &err);
+	if (dbus_error_is_set(&err)) {
+		dbus_error_free(&err);
+		dbus_connection_unref(fd.connection);
+		ERR_FAIL_V_MSG(FAILED, vformat("Failed to add DBus match: %s", err.message));
+	}
+
+	// Generate FileChooser message.
+	const char *method = nullptr;
+	switch (p_mode) {
+		case DisplayServer::FILE_DIALOG_MODE_SAVE_FILE: {
+			method = "SaveFile";
+		} break;
+		case DisplayServer::FILE_DIALOG_MODE_OPEN_ANY:
+		case DisplayServer::FILE_DIALOG_MODE_OPEN_FILE:
+		case DisplayServer::FILE_DIALOG_MODE_OPEN_DIR:
+		case DisplayServer::FILE_DIALOG_MODE_OPEN_FILES: {
+			method = "OpenFile";
+		} break;
+	}
+
+	DBusMessage *message = dbus_message_new_method_call(BUS_OBJECT_NAME, BUS_OBJECT_PATH, BUS_INTERFACE_FILE_CHOOSER, method);
+	{
+		DBusMessageIter iter;
+		dbus_message_iter_init_append(message, &iter);
+
+		append_dbus_string(&iter, p_xid);
+		append_dbus_string(&iter, p_title);
+
+		DBusMessageIter arr_iter;
+		dbus_message_iter_open_container(&iter, DBUS_TYPE_ARRAY, "{sv}", &arr_iter);
+
+		append_dbus_dict_string(&arr_iter, "handle_token", token);
+		append_dbus_dict_bool(&arr_iter, "multiple", p_mode == DisplayServer::FILE_DIALOG_MODE_OPEN_FILES);
+		append_dbus_dict_bool(&arr_iter, "directory", p_mode == DisplayServer::FILE_DIALOG_MODE_OPEN_DIR);
+		append_dbus_dict_filters(&arr_iter, p_filters);
+		append_dbus_dict_string(&arr_iter, "current_folder", p_current_directory, true);
+		if (p_mode == DisplayServer::FILE_DIALOG_MODE_SAVE_FILE) {
+			append_dbus_dict_string(&arr_iter, "current_name", p_filename);
+		}
+
+		dbus_message_iter_close_container(&iter, &arr_iter);
+	}
+
+	DBusMessage *reply = dbus_connection_send_with_reply_and_block(fd.connection, message, DBUS_TIMEOUT_INFINITE, &err);
+	dbus_message_unref(message);
+
+	if (!reply || dbus_error_is_set(&err)) {
+		dbus_error_free(&err);
+		dbus_bus_remove_match(fd.connection, fd.path.utf8().get_data(), &err);
+		dbus_connection_unref(fd.connection);
+		ERR_FAIL_V_MSG(FAILED, vformat("Failed to send DBus message: %s", err.message));
```

Comment:
```suggestion
		ERR_PRINT(vformat("Failed to send DBus message: %s", err.message));
		dbus_error_free(&err);
		dbus_bus_remove_match(fd.connection, fd.path.utf8().get_data(), &err);
		dbus_connection_unref(fd.connection);
		return FAILED;
```

Use `err` before freeing it. Also applies to other similar places.